frontend/src/features/properties/list/PropertyListView.tsx

```diff
@@ -18,27 +18,55 @@ const getPropertyListUrl = (filter: IPropertyFilter) =>
 const getPropertyReportUrl = (filter: IPropertyFilter) =>
   `${ENVIRONMENT.apiUrl}/reports/properties?${filter ? queryString.stringify(filter) : ''}`;
 
-const filter: IPropertyFilter = {
+const defaultFilter: IPropertyFilter = {
   page: 1,
-  quantity: 50,
+  quantity: 10,
   agencies: [1, 2, 3, 4, 5, 6],
 };
 
 const PropertyListView: React.FC = () => {
   const columns = useMemo(() => cols, []);
+
+  // We'll start our table without any data
   const [data, setData] = useState<IProperty[]>([]);
+  const [filter, setFilter] = useState<IPropertyFilter>({ ...defaultFilter });
+  // const [loading, setLoading] = useState(false);
+  const [pageCount, setPageCount] = useState(0);
+  const fetchIdRef = useRef(0);
+
+  // This will get called when the table needs new data
+  const fetchData = useCallback(
+    async ({ pageSize, pageIndex }: { pageSize: number; pageIndex: number }) => {
+      // Give this fetch an ID
+      const fetchId = ++fetchIdRef.current;
+
+      // TODO: Set the loading state
+      // setLoading(true);
+
+      // Only update the data if this is the latest fetch
+      if (fetchId === fetchIdRef.current) {
+        const newFilter = {
+          ...filter,
+          page: pageIndex + 1,
+          quantity: pageSize,
+        };
 
-  useEffect(() => {
-    const fetch = async () => {
-      try {
-        const result = await CustomAxios().get<IPagedItems<IProperty>>(getPropertyListUrl(filter));
-        setData(result.data.items);
-      } catch (e) {
-        setData([]);
+        const response = await CustomAxios().get<IPagedItems<IProperty>>(
+          getPropertyListUrl(newFilter),
+        );
+
+        // The server could send back total page count.
+        // For now we'll just calculate it.
+        setData(response.data.items);
+        setPageCount(Math.ceil(response.data.total / pageSize));
```

Comment:
This will technically be wrong if there is a remainder.

Reply:
That's why we use the math ceiling function. 15.2 will be rounded to 16